src/model/index.py

```diff
@@ -0,0 +1,18 @@
+import os
+from output.exceptions import *
+
+
+class Index:
+    def __init__(self, names, columns):
+        self.names = names
+        self.validate(self, columns)
+
+    @staticmethod
+    def validate(self, columns):
+        for name in self.names:
+            if name not in columns:
+                raise WrongParameterError("Index {} not found".format(name))
+
+    def write(self, path):
```

Comment:
rename it as `serialize`to follow the same name convention of index.serialize

Reply:
It shouldn’t take any path parameter.

You should say index.serialize()

you should pass the table object to the Index constructor. You are passing columns only.

how do you decide to pass TableMetadata in Index constructor or pass column array?

answer: you need to ask yourself, what is the relation between Table and Index?

There are four type of relations : -
- is a —> Student is a Person, so Student should inherit or extends from Person
- has many —> Table has many indeces, so table has an array of index
- has a —> Student has a bag, so student object  has one object of bag
- Belong to —> Index belongs to one Table, so Index has a Table object 

Here we can say Table has many indices and Every index belong to one table

Reply:
But shouldn't it know the table's path to be able to create the folders in it?